custom_components/NooLite.py

```diff
@@ -31,18 +29,13 @@
 
 DEVICE = None
 
-
 CONFIG_SCHEMA = vol.Schema({
     DOMAIN: vol.Schema({
         vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.string,
     }),
 }, extra=vol.ALLOW_EXTRA)
 
-
 PLATFORM_SCHEMA = vol.Schema({
-    vol.Required(CONF_NAME): cv.string,
-    vol.Required(CONF_CHANNEL): cv.positive_int,
-    vol.Required(CONF_MODE): cv.string,
```

Comment:
It is a really necessary to move this options to concrete modules? I use version ha 0.71 and it works without any issues.

Reply:
Have you tried to press "Check configuration" in Settings tab? vol.Required here indicates that Platform section in config should contain name, channel and mode properties. Indeed, it will work, but config validation will fail every time.

Reply:
Thanks. 
I have never used this possibility. You are right, the "Check configuration" returns error.